src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -234,62 +236,29 @@ class TextSentimentV4(override val textAnalyticsOptions: Option[TextAnalyticsReq
           WarningsV4(warnings.getMessage, warnings.getWarningCode.toString)))
     }
 
-    val textSentimentResult = textSentimentResultCollection.asScala.head
-    val documentSentiment = textSentimentResult.getDocumentSentiment();
-
-    val sentimentResult = if (textSentimentResult.isError) {
+    val sentimentResult = textSentimentResultCollection.map(sentiment => if(sentiment.isError) {
       None
     } else {
-      Some(getDocumentSentiment(documentSentiment))
-    }
+      Some(getDocumentSentiment(sentiment.getDocumentSentiment))
+    }).filter(_ != None).toList
 
-    val error = if (textSentimentResult.isError) {
-      val error = textSentimentResult.getError
-      Some(TAErrorV4(error.getErrorCode.toString, error.getMessage, error.getTarget))
-    } else {
-      None
-    }
+    val error = textSentimentResultCollection.map(sentiment => if (sentiment.isError) {
+      Some(TAErrorV4(sentiment.getError.getErrorCode.toString,
+              sentiment.getError.getMessage, sentiment.getError.getTarget))
+          } else {
+            None
+          }).filter(_ != None).toList

```

Comment:
similar here. can be used as:  collection.filter(sentiment => sentiment.isError).(==> transfer the error to TAErrorV4).